### PR TITLE
Don't pass `rng` to `FedimintModule`

### PR DESCRIPTION
**Pull request by dpc**
This is not going to work in modularization code.

### CODE REVIEW

File: fedimint-api/src/module/mod.rs [hunk 1]
```diff
@@ -142,7 +140,7 @@ pub struct ApiEndpoint<M> {
         for<'a> fn(&'a M, serde_json::Value) -> BoxFuture<'a, Result<serde_json::Value, ApiError>>,
 }
 
-#[async_trait(?Send)]
```
Comment (elsirion) [inline fedimint-api/src/module/mod.rs:145]: Do we need `Send`? Seems nice to have but broke WASM somehow? Although I'm wondering what `FederationModule` has to do with WASM :sweat_smile: Maybe we should just not compile the `FederationModule` impls in WASM mode for now. Eventually that will go into the server part of modules, which doesn't need WASM.

**Review by elsirion — APPROVED** [review]
ACK depending on CI fix

File: fedimint-api/src/module/mod.rs (hunk 1 above)
Comment (dpc) [reply fedimint-api/src/module/mod.rs:145]: Bummer. I was hoping to get rid of it, so starting consensus on another thread is easier...

Comment (elsirion) [reply fedimint-api/src/module/mod.rs]: I think to make it work you'd need to make a lot of stuff conditionally compiled if the target isn't WASM. Or find a way to make that WASM future `Send`.

**Review by elsirion — APPROVED** [review]

**Pull request merged**